server/datastore/mysql/hosts.go

```diff
@@ -771,33 +771,19 @@ func (ds *Datastore) MarkHostsSeen(ctx context.Context, hostIDs []uint, t time.T
 
 // SearchHosts performs a search on the hosts table using the following criteria:
 //	- Use the provided team filter.
-//	- Full-text search with the "query" argument (if query == "", then no fulltext matching is executed).
-// 	Full-text search is used even if "query" is a short or stopword.
-//	(what defines a short word is the "ft_min_word_len" VARIABLE, set to 4 by default in Fleet deployments).
+//	- Search hostname, uuid, hardware_serial, and primary_ip using LIKE (mimics ListHosts behavior)
 //	- An optional list of IDs to omit from the search.
-func (ds *Datastore) SearchHosts(ctx context.Context, filter fleet.TeamFilter, query string, omit ...uint) ([]*fleet.Host, error) {
-	var sqlb strings.Builder
-	sqlb.WriteString(`SELECT
+func (ds *Datastore) SearchHosts(ctx context.Context, filter fleet.TeamFilter, matchQuery string, omit ...uint) ([]*fleet.Host, error) {
+	query := `SELECT
 		h.*,
 		COALESCE(hst.seen_time, h.created_at) AS seen_time
 	FROM hosts h
 	LEFT JOIN host_seen_times hst
-	ON (h.id=hst.host_id) WHERE`)
+	ON (h.id=hst.host_id) WHERE TRUE `
 
 	var args []interface{}
-	if len(query) > 0 {
-		sqlb.WriteString(` (
-				MATCH (hostname, uuid) AGAINST (? IN BOOLEAN MODE)
-				OR MATCH (primary_ip, primary_mac) AGAINST (? IN BOOLEAN MODE)
-			) AND`)
-		// Transform query argument and append the truncation operator "*" for MATCH.
-		// From Oracle docs: "If a word is specified with the truncation operator, it is not
-		// stripped from a boolean query, even if it is too short or a stopword."
-		hostQuery := transformQueryWithSuffix(query, "*")
-		// Needs quotes to avoid each "." marking a word boundary.
-		// TODO(lucas): Currently matching the primary_mac doesn't work, see #1959.
-		ipQuery := `"` + query + `"`
-		args = append(args, hostQuery, ipQuery)
+	if len(matchQuery) > 0 {
+		query, args = hostSearchLike(query, args, matchQuery, hostSearchColumns...)
```

Comment:
Let's not forget to test performance of `Datastore.SearchHosts` in loadtest.

Reply:
It's pretty fast, you can test it right now: https://loadtest.fleetdm.com/hosts/manage/?query=--7n&order_key=hostname&order_direction=asc